app/enums/IncomeSourceJourney/IncomeSourceType.scala

```diff
@@ -16,28 +16,43 @@
 
 package enums.IncomeSourceJourney
 
+import forms.utils.SessionKeys
+
 sealed trait IncomeSourceType {
   val key: String
-  val addIncomeSourceStartDateMessagesPrefix: String
-  val addIncomeSourceStartDateCheckMessagesPrefix: String
+  val startDateMessagesPrefix: String
+  val addStartDateCheckMessagesPrefix: String
+  val startDateSessionKey: String
 }
 
 case object SelfEmployment extends IncomeSourceType {
   override val key = "SE"
-  override val addIncomeSourceStartDateMessagesPrefix = "add-business-start-date"
-  override val addIncomeSourceStartDateCheckMessagesPrefix = "add-business-start-date-check"
+  override val startDateMessagesPrefix = "add-business-start-date"
+  override val addStartDateCheckMessagesPrefix = "add-business-start-date-check"
+  override val startDateSessionKey = SessionKeys.addBusinessStartDate
 
 }
 
 case object UkProperty extends IncomeSourceType {
   override val key = "UK"
-  override val addIncomeSourceStartDateMessagesPrefix = "incomeSources.add.UKPropertyStartDate"
-  override val addIncomeSourceStartDateCheckMessagesPrefix = "add-uk-property-start-date-check"
+  override val startDateMessagesPrefix = "incomeSources.add.UKPropertyStartDate"
+  override val addStartDateCheckMessagesPrefix = "add-uk-property-start-date-check"
+  override val startDateSessionKey = SessionKeys.addUkPropertyStartDate
 }
 
 case object ForeignProperty extends IncomeSourceType {
   override val key = "FP"
-  override val addIncomeSourceStartDateMessagesPrefix = "incomeSources.add.foreignProperty.startDate"
-  override val addIncomeSourceStartDateCheckMessagesPrefix = "add-foreign-property-start-date-check"
+  override val startDateMessagesPrefix = "incomeSources.add.foreignProperty.startDate"
+  override val addStartDateCheckMessagesPrefix = "add-foreign-property-start-date-check"
+  override val startDateSessionKey = SessionKeys.foreignPropertyStartDate
+}
 
+object IncomeSourceType {
+  def get(key: String): IncomeSourceType = {
+    key match {
```

Comment:
if would guess we missing case where there is no match?
as we are converting String => IncomeSourceType

Reply:
you can change conversion method signature like:
`
object IncomeSourceType {
  def get(key: String): Option[IncomeSourceType] = {
    key match {
      case "FP" => Some(ForeignProperty)
      case "UK" => Some(UkProperty)
      case "SE" => Some(SelfEmployment)
      case _=> None
    }
  }
}
`

unless you can provide guaranties that key String will be one of these you pattern matching over, you would have to raise / log exception on some level